libraries/core_libs/network/src/tarcap/shared_states/syncing_state.cpp

```diff
@@ -8,40 +8,47 @@ namespace taraxa::network::tarcap {
 SyncingState::SyncingState(uint16_t deep_syncing_threshold)
     : malicious_peers_(300, 50), kDeepSyncingThreshold(deep_syncing_threshold) {}
 
-void SyncingState::set_peer(const dev::p2p::NodeID &peer_id) {
+void SyncingState::set_peer(const std::shared_ptr<TaraxaPeer> &peer) {
   std::unique_lock lock(peer_mutex_);
-  peer_id_ = peer_id;
+  peer_ = peer;
 }
 
 const dev::p2p::NodeID &SyncingState::syncing_peer() const {
   std::shared_lock lock(peer_mutex_);
-  return peer_id_;
+  return peer_->getId();
```

Comment:
is there a chance, when this can be nullptr?

Reply:
There shouldn't be, but I added the nullptr check anyway